Repository/ExtendedRepositories/RoleSystem/ActionPermissionRepository.cs

```diff
@@ -32,6 +33,14 @@ public Task AssignPermissionToAction(string ActionName, int PermissionId)
                 PermissionId = PermissionId
             });
         }
+
+        public void RemovePermissionFromAction(string actionName, string PermissionName)
+        {
+            var permissionOfAction = GetAll().Where(x =>
+                x.ActionName == actionName && x.PermissionId == PermissionsRepository.GetPermission(PermissionName).Id).FirstOrDefault();
+            SoftDelete(permissionOfAction);

```

Comment:
Use x.Permission.Name == PermissionName instead of calling GetPermission for a cleaner code and to perform the same task in a single query instead of two